modules/hostap/src/supp_api.c

```diff
@@ -906,10 +955,151 @@ int supplicant_channel(const struct device *dev, struct wifi_channel_info *chann
 }
 
 #ifdef CONFIG_AP
+#ifdef CONFIG_WIFI_NM_HOSTAPD_AP
+int hapd_state(const struct device *dev, int *state)
+{
+	struct hostapd_iface *iface;
+	int ret = 0;
+
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+	iface = get_hostapd_handle(dev);
+	if (!iface) {
+		wpa_printf(MSG_ERROR, "Device %s not found", dev->name);
+		ret = -1;
+		goto out;
+	}
+
+	*state = iface->state;
+
+out:
+	k_mutex_unlock(&wpa_supplicant_mutex);
+	return ret;
+}
+
+int hapd_config_network(struct hostapd_iface *iface,
```

Comment:
IIUC, the commands itself are common b/w SoftAP and FullAP, so, may be if you write a wrapper to choose wpa_cli_cmd_v vs hostapd_cli_cmd_v, we can reuse the code?

Reply:
Do you mean reuse the code of configuring network? I think the commands between softAP and fullAP are different.

Reply:
Yes.

> I think the commands between softAP and fullAP are different.

Ok, let me check the differences in commands then, thanks.